File: backend/app/schemas/case.py
import uuid
import datetime
from pydantic import BaseModel, ConfigDict
from app.models.enums import ContentType, Severity, CaseStatus

class CaseListResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id : uuid.UUID
    external_id : str
    severity : Severity
    status : CaseStatus
    content_type : ContentType
    source : str
    queue_id : uuid.UUID
    assigned_to : uuid.UUID | None
    sla_deadline : datetime.datetime | None

class CaseDetailResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    external_id: str
    content_type: ContentType
    content_ref: str
    content_snapshot: str | None
    severity: Severity
    status: CaseStatus
    queue_id: uuid.UUID
    assigned_to: uuid.UUID | None
    assigned_at: datetime.datetime | None
    sla_deadline: datetime.datetime | None
    source: str
    case_metadata: list
    created_at: datetime.datetime
    updated_at: datetime.datetime
    resolved_at: datetime.datetime | None

File: frontend/src/types/case.ts
export type CaseStatus = "unassigned" | "assigned" | "in_review" | "escalated" | "resolved";
export type Severity = 'low' | 'medium' | 'high' | 'critical';
export type ContentType = 'text' | 'image' | 'video' | 'audio' | 'account';

export interface Case {
    id: string;
    external_id: string;
    severity: Severity;
    status: CaseStatus;
    content_type: ContentType;
    source: string;
    queue_id: string;
    assigned_to: string | null;
    assigned_at: string | null;
    content_snapshot: string | null;
    created_at: string;
    updated_at: string;
    resolved_at: string | null;
    sla_deadline: string | null;
}